src/mesh/mesh_refinement.cpp

```diff
@@ -881,11 +931,31 @@ void MeshRefinement::ProlongateSharedFieldX2(
           Real gx3p = (coarse(k+1,j,i) - ccval)/dx3p;
           Real gx3c = 0.5*(SIGN(gx3m) + SIGN(gx3p))*std::min(std::abs(gx3m),
                                                              std::abs(gx3p));
-
-          fine(fk  ,fj,fi  ) = ccval - gx1c*(x1c - fx1m) - gx3c*(x3c - fx3m);
-          fine(fk  ,fj,fi+1) = ccval + gx1c*(fx1p - x1c) - gx3c*(x3c - fx3m);
-          fine(fk+1,fj,fi  ) = ccval - gx1c*(x1c - fx1m) + gx3c*(fx3p - x3c);
-          fine(fk+1,fj,fi+1) = ccval + gx1c*(fx1p - x1c) + gx3c*(fx3p - x3c);
+          if (std::strcmp(COORDINATE_SYSTEM, "spherical_polar") == 0) {
+            Real dfx1 = fx1p - fx1m;
+            Real dfx3 = fx3p - fx3m;
+            fine(fk  ,fj  ,fi) = ccval - gx1c*dfx1*(sarea_x2_[0][0](fi+1)
```

Comment:
This can cause 0/0 when the MeshBlock is touching the pole.
The easiest fix is just adding a small number to the denominator.